src/cocotb_tools/combine_results.py

```diff
@@ -1,110 +1,84 @@
-        "--testsuites_name",
-        dest="testsuites_name",
-        type=str,
-        required=False,
+        "--testsuites-name",
         default="results",
-        help="Name value for testsuites tag",
+        help="Name of 'testsuites' element in output XML file.",
     )
     parser.add_argument(
         "--verbose",
-        dest="debug",
-        action="store_const",
-        required=False,
-        const=True,
-        default=False,
-        help="Verbose/debug output",
+        action="store_true",
+        help="Enables verbose output",
     )
-    parser.add_argument(
-        "--suppress_rc",
-        dest="suppress_rc",
-        action="store_const",
-        required=False,
-        const=True,
-        default=False,
-        help="Suppress return code if failures found",
-    )
-
     return parser
 
 
-def main():
-    parser = get_parser()
+def main() -> int:
+    parser = _get_parser()
     args = parser.parse_args()
     rc = 0
 
     result = ET.Element("testsuites", name=args.testsuites_name)
 
-    for fname in find_all("results.xml", args.directory):
-        if args.debug:
-            print("Reading file %s" % fname)
-        tree = ET.parse(fname)
-        for ts in tree.iter("testsuite"):
-            if args.debug:
-                print(
-                    "Ts name : {}, package : {}".format(
-                        ts.get("name"), ts.get("package")
+    for directory in args.directories:
+        if args.verbose:
+            print(f"Searching in {directory} for results.xml files.")
+        for fname in _find_all("results.xml", directory):
+            if args.verbose:
+                print(f"Reading file {fname}.")
+            tree = ET.parse(fname)
+            for ts in tree.iter("testsuite"):
+                if args.verbose:
+                    print(
+                        "Testsuite name : {}, package : {}".format(
+                            ts.get("name"), ts.get("package")
+                        )
                     )
-                )
-            use_element = None
-            for existing in result:
-                if existing.get("name") == ts.get("name") and existing.get(
-                    "package"
-                ) == ts.get("package"):
-                    if args.debug:
-                        print("Already found")
-                    use_element = existing
-                    break
-            if use_element is None:
-                result.append(ts)
-            else:
-                # for tc in ts.getiterator("testcase"):
-                use_element.extend(list(ts))
-
-    if args.debug:
-        ET.dump(result)
+                use_element = None
+                for existing in result:
+                    if existing.get("name") == ts.get("name") and existing.get(
+                        "package"
+                    ) == ts.get("package"):
+                        if args.verbose:
+                            print("Already found.")
```

Comment:
What is this doing exactly?

Reply:
Saying whether the testsuite (whose info was just printed) was already found in the results. In that case it extends that testsuite rather than adding it "new" to the results.

Reply:
If the `name` and `package` of a `testsuite` has already been found, it merges the `testcase` elements into the previously found one.